permitchecklog.go

```diff
@@ -0,0 +1,204 @@
+package authaus
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/IMQS/log"
+	"sync"
+	"time"
+)
+
+// CheckLogEntry represents a single session check log entry
+type CheckLogEntry struct {
+	Timestamp    time.Time `json:"timestamp"`
+	SessionToken string    `json:"session_token"`
+	UserId       UserId    `json:"user_id"`
+}
+
+// CheckUsageTracker manages in-memory storage and periodic flushing of session check logs
+type CheckUsageTracker struct {
+	config           ConfigUsageTracking
+	log              *log.Logger
+	logs             []CheckLogEntry
+	mutex            sync.RWMutex
+	stopChan         chan struct{}
+	doneChan         chan struct{}
+	flushDoneChan    chan struct{}
+	stopOnce         sync.Once
+	flushTicker      *time.Ticker
+	flushing         bool // Track if a flush is in progress
+	db               *sql.DB
+	droppedLogsCount int64
```

Comment:
Maybe add droppedLogsCount to the database records, since there is no other exposure to it; it is only used in testing?